src/main/scala/li/cil/oc/server/PacketHandler.scala

```diff
@@ -220,6 +223,15 @@ object PacketHandler extends CommonPacketHandler {
     }
   }
 
+  def onDatabaseSetSlot(p: PacketParser) {
+    val slot = p.readByte()
+    val stack = p.readItemStack()
+    p.player.openContainer match {
+      case db: Database => db.putStackInSlot(slot, stack)
```

Comment:
You should validate the slot index. Otherwise hackers would forge a slot index that maps to player inventory and start duping stuff.

Reply:
huh, hackers, ok